client/my-sites/stats/activity-log-item/activity-title.jsx

```diff
@@ -0,0 +1,289 @@
+				url: PropTypes.shape( {
+					host: PropTypes.string.isRequired,
+					url: PropTypes.string.isRequired,
+					host_reversed: PropTypes.string.isRequired,
+				} ).isRequired,
+			} ),
+
+			comment: PropTypes.shape( {
+				approved: PropTypes.bool.isRequired,
+				id: PropTypes.number.isRequired,
+			} ),
+
+			core: PropTypes.shape( {
+				new_version: PropTypes.string,
+				old_version: PropTypes.string,
+			} ),
+
+			plugin: PropTypes.oneOfType( [
+				PropTypes.shape( {
+					name: PropTypes.string,
+					previous_version: PropTypes.string,
+					slug: PropTypes.string,
+					version: PropTypes.string,
+				} ),
+				PropTypes.arrayOf(
+					PropTypes.shape( {
+						name: PropTypes.string,
+						previous_version: PropTypes.string,
+						slug: PropTypes.string,
+						version: PropTypes.string,
+					} ),
+				),
+			] ),
+
+			post: PropTypes.shape( {
+				id: PropTypes.number.isRequired,
+				status: PropTypes.string.isRequired,
+				type: PropTypes.string,
+				title: PropTypes.string,
+			} ),
+
+			term: PropTypes.shape( {
+				id: PropTypes.number.isRequired,
+				title: PropTypes.string.isRequired,
+				type: PropTypes.string.isRequired,
+			} ),
+
+			theme: PropTypes.oneOfType( [
+				PropTypes.arrayOf(
+					PropTypes.shape( {
+						name: PropTypes.string,
+						slug: PropTypes.string,
+						uri: PropTypes.string,
+						version: PropTypes.string,
+					} )
+				),
+				PropTypes.shape( {
+					name: PropTypes.string,
+					slug: PropTypes.string,
+					uri: PropTypes.string,
+					version: PropTypes.string,
+				} ),
+			] ),
+
+			user: PropTypes.shape( {
+				display_name: PropTypes.string,
+				external_user_id: PropTypes.string,
+				login: PropTypes.string,
+				wpcom_user_id: PropTypes.number,
+			} ),
+
+			widget: PropTypes.shape( {
+				id: PropTypes.number,
+				name: PropTypes.string,
+				sidebar: PropTypes.string,
+			} ),
+		} ),
+
+		// localize
+		moment: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	getAction() {
+		return this.props.action || this.props.name;
+	}
+
+	getActorName() {
+		const { actor } = this.props;
+		const displayName = get( actor, 'display_name' );
+		if ( displayName ) {
+			return displayName;
+		}
+		const login = get( actor, 'display_name' );
+		if ( login ) {
+			return login;
+		}
+
+		return 'An unknown user';
+	}
+
+	getUserName() {
+		const user = get( this.props.object, 'user' );
+		const displayName = get( user, 'display_name' );
+		if ( displayName ) {
+			return displayName;
+		}
+		const login = get( user, 'display_name' );
+		if ( login ) {
+			return login;
+		}
+
+		return 'an unknown user';
+	}
+
+	getPluginName() {
+		const pluginObject = get( this.props.object, 'plugin' );
+		const getName = plugin => {
+			const name = get( plugin, 'name' );
+			if ( name ) {
+				return name;
+			}
+			const slug = get( plugin, 'slug' );
+			if ( slug ) {
+				return slug;
+			}
+			return 'an unknown plugin';
```

Comment:
That's what I thought, but `get` returns a defined property even if it's falsey. Most of these props are set to `""`, so I actually need to check the falsey-ness.